site/src/self_profile.rs

```diff
@@ -240,19 +240,20 @@ async fn download_and_analyze_self_profile(
 /// backend, linker) from the self-profile queries.
 fn compute_compilation_sections(profile: &ProfilingData) -> Vec<CompilationSection> {
     let mut first_event_start = None;
-    let mut frontend_end = None;
     let mut backend_start = None;
     let mut backend_end = None;
+    let mut rust_codegen = None;
     let mut linker_duration = None;
 
     for event in profile.iter_full() {
         if first_event_start.is_none() {
             first_event_start = event.payload.timestamp().map(|t| t.start());
         }
 
-        if event.label == "analysis" {
-            // End of "analysis" => end of frontend
-            frontend_end = event.payload.timestamp().map(|t| t.end());
+        if event.label == "collect_and_partition_mono_items" {
+            // We mark this query as "Rust codegen", where monomorphization and MIR-to-backend
+            // conversion happens.
```

Comment:
collect_and_partition_mono_items happens before the translation from MIR to llvm ir. It is the step which determines which instances to actually codegen and into which codegen unit. I believe it also implicitly loads the MIR for dependencies.

Reply:
Aah, you're right, thanks. The MIR-to-IR translation happens in `codegen_module`, right? That explains the staircase shape:
![image](https://github.com/rust-lang/rustc-perf/assets/4539057/b05a6a6b-3ed3-4bb6-8df7-3678aed9e458)

Hmm, then this becomes quite tricky, as the MIR-to-IR computation is overlaid with LLVM, and we can no longer render it as a separate part.

I still think that the mono collector is a reasonably self-contained part and it could be useful to visualize it (with a more accurate description, of course). What do you think?

Reply:
> The MIR-to-IR translation happens in codegen_module, right?

Yes

> Hmm, then this becomes quite tricky, as the MIR-to-IR computation is overlaid with LLVM, and we can no longer render it as a separate part.

If you make the graph represent cpu-clock (time spent on each individual thread added together) rather than task-clock or wall-time you can add all `codegen_module` queries together to form "Rust codegen" time and all `codegen_module_optimize` queries to form "backend" time.

Reply:
`cpu-clock` is not supported by `self-profile`, and also if we added time of multiple threads together, it would no longer make sense to show this in the same chart as a fraction of the whole compilation time (it could be another chart, but that's probably overkill).